Parsers/ASimAuthentication/Parsers/vimAuthenticationIllumioSaaSCore.yaml

```diff
@@ -0,0 +1,143 @@
+Normalization:
+  Schema: Authentication
+  Version: '0.1.3'
+References:
+  - Title: ASIM Authentication Schema
+    Link: https://aka.ms/ASimAuthenticationDoc
+  - Title: ASIM
+    Link: https://aka.ms/AboutASIM
+Description: |
+  This ASIM parser supports normalizing Illumio sign in logs, stored in the Illumio_Auditable_Events_CL table, to the ASIM Authentication schema. 
+ParserName: vimAuthenticationIllumioSaaSCore
+EquivalentBuiltInParser: _Im_Authentication_IllumioSaaSCore
+ParserParams:
+  - Name: starttime
+    Type: datetime
+    Default: datetime(null)
+  - Name: endtime
+    Type: datetime
+    Default: datetime(null)
+  - Name: username_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: targetappname_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: srcipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: srchostname_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: eventtype_in
+    Type: dynamic
+    Default: dynamic([])
+  - Name: eventresultdetails_in
+    Type: dynamic
+    Default: dynamic([])
+  - Name: eventresult
+    Type: string
+    Default: '*'
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+    let EventTypeLookup = datatable(
+        event_type: string, // what Illumio sends
+        EventType: string, // an enumerated list [ Logon, Logoff, Elevate ] event type
+        EventResultDetails: string,
+        EventResult: string
+    )
+    [
+        'user.authenticate', 'Logon', 'Other', 'Success',
+        'user.login', 'Logon', 'Other', 'Success',
+        'user.logout', 'Logoff', 'Other', 'Success',
+        'user.sign_in', 'Logon', 'Other', 'Success',
+        'user.sign_out', 'Logoff', 'Other', 'Success',
+        'user.use_expired_password', 'Logon', 'Password expired', 'Success'
+    ];
+    let parser=(
+        starttime: datetime=datetime(null), 
+        endtime: datetime=datetime(null), 
+        username_has_any: dynamic = dynamic([]),
+        targetappname_has_any: dynamic = dynamic([]),
+        srcipaddr_has_any_prefix: dynamic = dynamic([]),
+        srchostname_has_any: dynamic = dynamic([]),
+        eventtype_in: dynamic = dynamic([]),
+        eventresultdetails_in: dynamic = dynamic([]),
+        eventresult: string = '*',
+        disabled: bool=false
+    ) {
+      Illumio_Auditable_Events_CL
+      | where not(disabled) and  event_type startswith "user" and event_type !endswith "terminated" // limited to user signin, login, logoff, signoff events only
+      | where
+          (isnull(starttime) or TimeGenerated >= starttime) 
+          and (isnull(endtime) or TimeGenerated <= endtime) 
+          and (array_length(targetappname_has_any) == 0) // TargetAppName not available in source
+          and (array_length(srchostname_has_any) == 0)  // srchostname_has_any not available in source          
+          and (array_length(eventresultdetails_in) == 0) // EventResultDetails not available in source
```

Comment:
You are setting eventresultdetails_in. Please update the filter condition post doing lookup 'EventTypeLookup' 
| where (array_length(eventresultdetails_in) == 0 or EventResultDetails in~ (eventresultdetails_in))